python/cudf/cudf/core/column/column.py

```diff
@@ -1042,24 +1132,30 @@ def copy(self, deep: bool = True) -> Self:
                 )
             return result._with_type_metadata(self.dtype)  # type: ignore[return-value]
         else:
-            return cast(
-                Self,
-                build_column(
```

Comment:
How close are we to `build_column` going away altogether? We've been making slow and steady progress towards using column constructors instead, there can't be much left now.